eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelFeatures.scala

```diff
@@ -31,26 +30,20 @@ import fr.acinq.eclair.{Feature, FeatureSupport, Features}
  */
 case class ChannelFeatures(activated: Set[Feature]) {
 
-  /** Format of the channel transactions. */
-  val commitmentFormat: CommitmentFormat = {
-    if (hasFeature(AnchorOutputs)) {
-      AnchorOutputsCommitmentFormat
-    } else {
-      DefaultCommitmentFormat
-    }
-  }
-
   val channelType: SupportedChannelType = {
-    if (hasFeature(AnchorOutputs)) {
+    if (hasFeature(Features.AnchorOutputsZeroFeeHtlcTxs)) {
+      ChannelTypes.AnchorOutputsZeroFeeHtlcTxs
+    } else if (hasFeature(Features.AnchorOutputs)) {
       ChannelTypes.AnchorOutputs
-    } else if (hasFeature(StaticRemoteKey)) {
+    } else if (hasFeature(Features.StaticRemoteKey)) {
       ChannelTypes.StaticRemoteKey
     } else {
       ChannelTypes.Standard
     }
   }
 
   val paysDirectlyToWallet: Boolean = channelType.paysDirectlyToWallet
+  val commitmentFormat: CommitmentFormat = channelType.commitmentFormat
```

Comment:
For the sake of discussing: why not define two commitment formats for anchor, instead of passing around the `channelType`? Things like below make me doubt, because the commitment format should be what we use to deal with commitment-related stuff:

```scala
val commitFees = commitTxTotalCostMsat(remoteParams.dustLimit, reduced, channelType)
```

Could be:

```scala
sealed trait AnchorOutputsCommitmentFormat

case object UnsafeLegacyAnchorOutputsCommitmentFormat extends AnchorOutputsCommitmentFormat

case object ZeroFeeHtlcAnchorOutputsCommitmentFormat extends AnchorOutputsCommitmentFormat
```

Reply:
I view `commitmentFormat` as `the format of the commit tx`, and it's the same in both cases...but I agree that the `channelType` felt a bit redundant, I'll try that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1932/commits/c537fcb1c68e708e0c258f7787194ee4d33ead45, it provides a cleaner separation between the `transactions` package and the `channel` package, transactions don't need to receive a channel type.